code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -0,0 +1,214 @@
+			return TRUE
+		if(istype(the_target, /mob/living/simple_animal/hostile/rat) && A.stat == CONSCIOUS)
+			var/mob/living/simple_animal/hostile/rat/R = the_target
+			if(R.faction_num == faction_num)
+				return FALSE
+			else
+				return TRUE
+		return ..()
+
+/**
+  *This action creates trash, money, dirt, and cheese.
+  */
+/datum/action/cooldown/coffer
+	name = "Fill Coffers"
+	desc = "Your newly granted regality and poise let you scavenge for lost junk, but more importantly, cheese."
+	icon_icon = 'icons/mob/actions/actions_animal.dmi'
+	background_icon_state = "bg_clock"
+	button_icon_state = "coffer"
+	cooldown_time = 50
+
+/datum/action/cooldown/coffer/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/turf/T = get_turf(owner)
+	var/loot = rand(1,100)
+	switch(loot)
+		if(1 to 5)
+			to_chat(owner, "<span class='notice'>Score! You find some cheese!</span>")
+			new /obj/item/reagent_containers/food/snacks/cheesewedge(T)
+		if(6 to 10)
+			var/pickedcoin = pick(GLOB.ratking_coins)
+			to_chat(owner, "<span class='notice'>You find some leftover coins. More for the royal treasury!</span>")
+			for(var/i = 1 to rand(1,3))
+				new pickedcoin(T)
+		if(11)
+			to_chat(owner, "<span class='notice'>You find a... Hunh. This coin doesn't look right.</span>")
+			var/rarecoin = rand(1,2)
+			if (rarecoin == 1)
+				new /obj/item/coin/twoheaded(T)
+			else
+				new /obj/item/coin/antagtoken(T)
+		if(12 to 40)
+			var/pickedtrash = pick(GLOB.ratking_trash)
+			to_chat(owner, "<span class='notice'>You just find more garbage and dirt. Lovely, but beneath you now.</span>")
+			new /obj/effect/decal/cleanable/dirt(T)
+			new pickedtrash(T)
+		if(41 to 100)
+			to_chat(owner, "<span class='notice'>Drat. Nothing.</span>")
+			new /obj/effect/decal/cleanable/dirt(T)
+	StartCooldown()
+
+/**
+  *This action checks all nearby mice, and converts them into hostile rats. If no mice are nearby, creates a new one.
+  */
+
+/datum/action/cooldown/riot
+	name = "Raise Army"
+	desc = "Raise an army out of the hordes of mice and pests crawling around the maintenance shafts."
+	icon_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "riot"
+	background_icon_state = "bg_clock"
+	cooldown_time = 80
+	///Checks to see if there are any nearby mice. Does not count Rats.
+
+/datum/action/cooldown/riot/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/cap = CONFIG_GET(number/ratcap)
+	var/something_from_nothing = FALSE
+	for(var/mob/living/simple_animal/mouse/M in oview(owner, 5))
+		var/mob/living/simple_animal/hostile/rat/new_rat = new(get_turf(M))
+		something_from_nothing = TRUE
+		if(M.mind && M.stat == CONSCIOUS)
+			M.mind.transfer_to(new_rat)
+		if(istype(owner,/mob/living/simple_animal/hostile/regalrat))
+			var/mob/living/simple_animal/hostile/regalrat/giantrat = owner
+			new_rat.faction_num = giantrat.faction_num
+		qdel(M)
+	if(!something_from_nothing && LAZYLEN(SSmobs.cheeserats) >= cap)
```

Comment:
Make this check as a child of the next one, checking `if(! something_from_nothing)` just once